src/uct/base/uct_md.c

```diff
@@ -346,8 +342,9 @@ ucs_status_t uct_rkey_unpack(uct_component_h component, const void *rkey_buffer,
 {
     char mdc_name[UCT_MD_COMPONENT_NAME_MAX + 1];
 
-    if (ENABLE_PARAMS_CHECK && ENABLE_DEBUG_DATA) {
-        if (strncmp(rkey_buffer, component->name, UCT_MD_COMPONENT_NAME_MAX)) {
+    if (ENABLE_DEBUG_DATA) {
```

Comment:
given the change in uct_md_mkey_pack, shouldn't we do ' rkey_buffer += UCT_MD_COMPONENT_NAME_MAX' unconditionally now?

Reply:
no, the logic left the same, the check was moved to the common function